Wire version detect master

jfriesse opened the pull request:
Add magic and version into header

Magic number (0xC070) together with version in every packet
is used for detecting that other node is really
Corosync 3.x.

Endian_detector field is removed and magic number is now
used instead.

If received packet magic number differs, guessing is used to show more
about the source (Corosync 2.3+, 2.2 are quite reliable, Knet and
unencrypted Corosync 2.1/2.0/1.x/OpenAIS are semi-reliable and encrypted
Corosync 2.1/2.0/1.x/OpenAIS are quite unreliable).

chrissie-c commented:
Tests fine for me. it's also about time we put the system_from in that message :)

jfriesse commented:
@chrissie-c Thank you for the review. I've completely forgot to test encrypted Knet so I've changed one of the message to be able to detect also that condition.

I've also changed struct sockaddr_storage passing from struct to const pointer (just to save some bytes in the stack).

chrissie-c commented:
ACK. works for me.

jfriesse commented:
@chrissie-c Thank you for the review. I've merged patch as a 69857efb5b24dc2473acc6e7653e5e41886ad03d.
